lib/spree/core/controller_helpers/order.rb

```diff
@@ -68,18 +68,9 @@ def associate_user
           session[:guest_token] = nil
         end
 
-        # Recover incomplete orders from other sessions after logging in.
+        # Load current order and create a new one if necessary.
         def set_current_order
-          return unless (user = spree_current_user)
-
-          last_incomplete_order = user.last_incomplete_spree_order
-
-          if session[:order_id].nil? && last_incomplete_order
```

Comment:
I am not sure about this statement:
"Every page load creates a cart order if none is present."

I thought this method was the place where that was happening, right?
I thought that this conditional was recovering existing carts if no order is identified in the session.
By doing this I think we are just increasing the number of past carts for the users and even maybe loosing resetting the cart between sessions.

My test would be: login to OFN and add something to the cart.
Then, open an icognito tab and login again, do you see your cart as it was?

Reply:
Yeah, my hypothesis is confirmed by my test.
I run this test above in master and with this PR and in master the cart is preserved into the incognito tab after login, with this PR, after login in the incognito tab, the cart is empty.
We need to revert this change and keep this conditional.

Reply:
I reverted the last commit so we can move this PR forward now.

Reply:
Sounds good to me

Reply:
ok, awesome, moving to Test Ready now.

Reply:
Okay, great. I tried to test that and I couldn't find a way to recover an old cart. Can you share your steps?

Reply:
just these as described above:
- login to OFN and add something to the cart.
- Then, open an icognito tab and login again, do you see your cart as it was?